ethylene-core/src/main/java/com/github/steanky/ethylene/core/ConfigElement.java

```diff
@@ -128,78 +141,78 @@ default boolean isNull() {
      * Determines if this ConfigElement represents an object. This is true for {@link ConfigPrimitive} and should be
      * true for specialized, direct implementations of this interface that do not, themselves, hold on to ConfigElement
      * instances. It should be false for {@link ConfigNode} and {@link ConfigList}.
-     * @return true if {@link ConfigElement#asObject()} will succeed without throwing an exception; false otherwise
+     *
+     * @return true if {@link ConfigElement#asScalar()} will succeed without throwing an exception; false otherwise
      */
-    default boolean isObject() {
+    default boolean isScalar() {
         return false;
     }
 
     /**
-     * Converts this ConfigElement into an object. For elements whose purpose it is to "wrap" Java types, this method
-     * should return the wrapped object. Although it may seem counterintuitive, this method <i>should</i> throw an
-     * exception for {@link ConfigList} and {@link ConfigNode} implementations — because these types are not meant to be
-     * simple wrappers.
+     * Converts this ConfigElement into the <i>scalar</i> Java type it represents. Scalar types are types that cannot
+     * themselves contain additional ConfigElements. In Ethylene Core, the only scalar ConfigElement implementation is
+     * {@link ConfigPrimitive}. Other modules may add scalar types specific to certain formats.
+     *
      * @return this element as an object
      * @throws IllegalStateException if this element cannot be converted into an object
      */
-    default Object asObject() { throw new IllegalStateException("Element may not be converted to Object"); }
+    default Object asScalar() {throw new IllegalStateException("Element may not be converted to Object");}
```

Comment:
where does this word come from

Reply:
https://yaml.org/spec/1.2.2/#scalar

Reply:
nobody likes yaml

Reply:
i believe the term is used elsewhere as well
i changed it from "object" since object is a bit confusing (everything is an object)